test/worlds/imu_heading_deg.sdf

```diff
@@ -0,0 +1,132 @@
+<?xml version="1.0" ?>
+<sdf version="1.6">
+  <world name="imu_sensor">
+    <gravity>0 0 -5</gravity>
+    <physics name="1ms" type="ode">
+      <max_step_size>0.001</max_step_size>
+      <real_time_factor>0</real_time_factor>
+    </physics>
+    <plugin
+      filename="ignition-gazebo-physics-system"
+      name="ignition::gazebo::systems::Physics">
+    </plugin>
+    <plugin
+      filename="ignition-gazebo-imu-system"
+      name="ignition::gazebo::systems::Imu">
+    </plugin>
+
+    <spherical_coordinates>
+      <heading_deg>90</heading_deg>
+    </spherical_coordinates>
+
+    <model name="ground_plane">
+      <static>true</static>
+      <link name="link">
+        <collision name="collision">
+          <geometry>
+            <plane>
+              <normal>0 0 1</normal>
+              <size>100 100</size>
+            </plane>
+          </geometry>
+        </collision>
+        <visual name="visual">
+          <geometry>
+            <plane>
+              <normal>0 0 1</normal>
+              <size>100 100</size>
+            </plane>
+          </geometry>
+          <material>
+            <ambient>0.8 0.8 0.8 1</ambient>
+            <diffuse>0.8 0.8 0.8 1</diffuse>
+            <specular>0.8 0.8 0.8 1</specular>
+          </material>
+        </visual>
+      </link>
+    </model>
+
+    <model name="imu_model">
+      <pose>4 0 3.0 0 0.0 3.14</pose>
```

Comment:
@adityapande-1995 can we add some tests for IMUs with rotated link and/or sensor frames? I've observed some odd behavior in the DEFAULT case when the sensor frame attempts to match NED and the link frame is rotated in yaw.

Reply:
I'm not sure I understood that correctly. So there should be a test where the "link" has some non zero pose, and the sensor (model) rotates to match some orientation (say NED), and that should be reported correctly ?

Reply:
> So there should be a test where the "link" has some non zero pose, and the sensor (model) rotates to match some orientation (say NED), and that should be reported correctly.

@adityapande-1995 that's right. More generally, my suggestion is we test a few combinations of non-identity link and sensor frame rotations (as opposed to one) to make sure this is working as intended.